### PR TITLE
feat: Implement verify method

### LINKED ISSUE #20: Auth#verifyIdToken always throws unimplemented error
Version 0.3.0

I know it's an early build, so just drawing attention to a blocking feature for wider adoption. We use `verifyIdToken` to verify requests from clients before performing requests using admin functionality or firestore. 

Unfortunately, I don't have enough familiarity with the jwt spec and library design to understand what the `jwtKey` and other missing params are at `src/utils/jwt.dart:100`.

Issue:
Auth#verifyIdToken always throws unimplemented error.

Reproduce:
```
final FirebaseAdminApp admin =  await FirebaseAdminInitialization(container).initialize();
final auth = Auth(admin);
final idToken = await auth.verifyIdToken(jwt);
await admin.close();
```

StackTrace:
```
#0      PublicKeySignatureVerifier.verify (package:dart_firebase_admin/src/utils/jwt.dart:100)
#1      FirebaseTokenVerifier._verifySignature (package:dart_firebase_admin/src/auth/token_verifier.dart:119)
#2      FirebaseTokenVerifier._decodeAndVerify (package:dart_firebase_admin/src/auth/token_verifier.dart:101)
<asynchronous suspension>
#3      FirebaseTokenVerifier.verifyJWT (package:dart_firebase_admin/src/auth/token_verifier.dart:76)
<asynchronous suspension>
#4      _BaseAuth.verifyIdToken (package:dart_firebase_admin/src/auth/base_auth.dart:360)
```

### PR DESCRIPTION
**Pull request by KKimj**
## Related Issues

fixes #20 #33 
Thanks to @tkfinitefield 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).

- [ ] I have updated the `CHANGELOG.md` of the relevant packages.
      Changelog files must be edited under the form:

  ```md
  ## Unreleased fix/major/minor

  - Description of your change. (thanks to @yourGithubId)
  ```

- [x] If this contains new features or behavior changes,
      I have updated the documentation to match those changes.

### CODE REVIEW

**Comment by rrousselGit** [comment]
As mentioned in the original PR, we'd want tests :)

**Comment by KKimj** [comment]
@rrousselGit 

I'm having difficulty designing test scenarios when trying to write test code.
I think testing the process of encoding and decoding a dummy JWT token might be a good approach, but I'm curious to hear your thoughts.

If you have the time, I'd be interested in collaborating on writing the tests.

File: packages/dart_firebase_admin/lib/src/utils/jwt.dart [hunk 1]
```diff
@@ -96,9 +96,40 @@ class PublicKeySignatureVerifier implements SignatureVerifier {
...
+      final publicKeys = await keyFetcher.fetchPublicKeys();
+      final publicKey = publicKeys[kid];
+
+      if (publicKey == null) {
+        throw JwtError(
+          JwtErrorCode.noMatchingKid,
+          'no-matching-kid-error',
+        );
+      }
+      JWT.verify(
```
Comment (labrom) [inline packages/dart_firebase_admin/lib/src/utils/jwt.dart:120]: Looking at what needs to be verified from https://firebase.google.com/docs/auth/admin/verify-id-tokens#verify_id_tokens_using_a_third-party_jwt_library:
- audience: already verified in FirebaseTokenVerifier._verifyContent()
- issuer: already verified in FirebaseTokenVerifier._verifyContent()
- sub: presence already verified in FirebaseTokenVerifier._verifyContent()
- exp: unless I missed something, doesn't seem to be verified, but JWT.verify() will take care of it (this should be documented because for the sake of consistency we would have wanted to verify it in _verifyContent())
- iat: doesn't seem to be verified either, JWT.verify() has the issueAt parameter but the way it handles it and verifies the token's iat confuses me, so I would actually recommend checking it in _verifyContent()

So in summary, the use of JWT.verify() without any parameter in the proposed implementation is consistent with the existing implementation, assuming 'iat' is verified in _verifyContent().

I would recommend removing the existing verifyJwtSignature() function and the commented-out line that invokes it.

**Comment by oikkoikk** [comment]
are there any updates?

**Comment by labrom** [comment]
I couldn't modify this PR so I created a new one: https://github.com/invertase/dart_firebase_admin/pull/54.
The changes are mostly copied over except I tried to reuse the existing `verifyJwtSignature` function.
The unit tests verify the 'kid' handling that is added to the `verify` function.

**Comment by rrousselGit** [comment]
Closing since there's a new PR
